### PR TITLE
chore(style): use lv_part_t type where suitable

### LINKED ISSUE #6065: parameter types of functions in lv_obj_style_gen.h are incorrect
### Introduce the problem

```c
static inline int32_t lv_obj_get_style_width(const lv_obj_t * obj, uint32_t part)
{
    lv_style_value_t v = lv_obj_get_style_prop(obj, part, LV_STYLE_WIDTH);
    return (int32_t)v.num;
}

```

the function that gets the property is correctly declared

```c
lv_style_value_t lv_obj_get_style_prop(const lv_obj_t * obj, lv_part_t part, lv_style_prop_t prop);
```

### Proposal

Should be

```c
static inline int32_t lv_obj_get_style_width(lv_obj_t * obj, lv_part_t part)
{
    lv_style_value_t v = lv_obj_get_style_prop(obj, part, LV_STYLE_WIDTH);
    return (int32_t)v.num;
}
```

### PR DESCRIPTION
**Pull request by liamHowatt**
### Description of the feature or fix

Fixes #6065

For consistency, use the typedef `lv_part_t` instead of `uint32_t` where a part variable is declared.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Review by W-Mai — APPROVED** [review]

**Review by FASTSHIFT — DISMISSED** [review]

File: docs/overview/draw.rst [hunk 1]
```diff
@@ -295,7 +295,7 @@ documentation.
         const lv_point_t * p2;        /**< A point calculated during drawing. E.g. a point of chart.*/
         char * text;                  /**< A text calculated during drawing. Can be modified. E.g. tick labels on a chart axis.*/
         uint32_t text_length;         /**< Size of the text buffer containing null-terminated text string calculated during drawing.*/
-        uint32_t part;                /**< The current part for which the event is sent*/
+        lv_part_t part;                /**< The current part for which the event is sent*/
```
Comment (kisvegabor) [inline docs/overview/draw.rst]: This page of the docs is for v8 and outdated and mostly [this](https://docs.lvgl.io/master/porting/draw.html#draw-task) is used.  Please remove this file instead, and we can add a new one later.

**Review by kisvegabor — COMMENTED** [review]
Looks good, just one note.

**Review by kisvegabor — COMMENTED** [review]
`draw.rst` shall be removed from [index.rst](https://github.com/lvgl/lvgl/blob/master/docs/overview/index.rst) too.

**Review by kisvegabor — APPROVED** [review]
Thank you!

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**